Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -870,19 +870,35 @@ class BrowserViewController: UIViewController {
     }
 
     func finishEditingAndSubmit(_ url: URL, visitType: VisitType) {
-        topToolbar.currentURL = url
-        topToolbar.leaveOverlayMode()
+        let javascriptPrefix = "javascript:"
+        if url.absoluteString.hasPrefix(javascriptPrefix) {
```

Comment:
I feel like we should add `isBookmarklet` helper to `URL`, add it to `URLExtensions.swift` and replace all these hasPrefix occurences